packages/flutter/lib/src/cupertino/tab_scaffold.dart

```diff
@@ -156,29 +243,50 @@ class CupertinoTabScaffold extends StatefulWidget {
 }
 
 class _CupertinoTabScaffoldState extends State<CupertinoTabScaffold> {
-  int _currentPage;
+  CupertinoTabController _controller;
 
   @override
   void initState() {
     super.initState();
-    _currentPage = widget.tabBar.currentIndex;
+    _updateTabController();
   }
 
+  void _updateTabController() {
+    final CupertinoTabController newController =
+      // User provided a new controller, update [_controller] with it.
+      widget.controller
+      // User didn't provide a new controller, keep using [_controller].
+      ?? _controller
```

Comment:
I wouldn't do this clause. If the user specified a controller and then gave null, we should just go back to tab 0, or whatever tabBar.currentIndex is.